### DISCUSSION TITLE
Custom header for webhooks not working

### DISCUSSION THREAD

**Discussion by ishita-chakraborty** [netbox-community/netbox#9355] [Q&A]
I was trying to pass a custom headers for the webhook setup and it is not working. Can someone provide a picture of how the additional header row be populated in the webhook configuration form and if it works?

**Comment by candlerb** [comment]
Yes it works - <a href="http://www.catb.org/~esr/faqs/smart-questions.html#idm368">reference</a>.

Just put your custom header and value in the box in the format "Name: Value", exactly like the help text tells you to do:

![image](https://user-images.githubusercontent.com/44789/168158639-a7ee7fd5-2569-4b9d-885e-e7c27ea1ccb5.png)

I think you should approach the problem the other way round:

1. What version of Netbox are you running?
2. How exactly have you configured the webhook at the Netbox side?
3. What exactly is arriving at the target server, for example as viewed using tcpdump?

**Reply by ishita-chakraborty** [reply]
1. I am running 2.11.7. 
2.  Attaching screenshots for the webhook configuration
3.  The target server receives everything except the headers.
![image](https://user-images.githubusercontent.com/84405532/168179418-d38a05f7-4a5a-4ef7-a656-364633f2e62f.png)
![image](https://user-images.githubusercontent.com/84405532/168179701-56844bcf-4bab-44fb-9268-b7ef2a3ab81a.png)

**Reply by candlerb** [reply]
It works for me - I'm using Netbox 3.2.3, and I can see the additional headers in tcpdump.

My guess is that your target server is rejecting header names which contain an underscore.  They are not technically illegal, but frowned upon, and many web servers drop them.  See:
* https://stackoverflow.com/questions/22856136/why-do-http-servers-forbid-underscores-in-http-header-names
* https://www.grouparoo.com/blog/dont-use-underscores-in-http-headers

**Reply by ishita-chakraborty** [reply]
That worked! Thanks a lot!
